### DISCUSSION TITLE
[Docs]: KRaft Limitations

### DISCUSSION THREAD

**Discussion by Pythoner6** [strimzi/strimzi-kafka-operator#9702] [Q&A] [answered]
### Suggestion / Problem

The docs currently say the following about KRaft (emphasis mine)

> Currently, the KRaft mode in Strimzi has the following major limitations:
> - Moving from Kafka clusters with ZooKeeper to KRaft clusters or the other way around is not supported.
> - **Controller-only nodes cannot undergo rolling updates or be updated individually.**
> - **Upgrades and downgrades of Apache Kafka versions or the Strimzi operator are not supported. Users might need to delete the cluster, upgrade the operator and deploy a new Kafka cluster.**
> - Only the Unidirectional Topic Operator is supported in KRaft mode. The Bidirectional Topic Operator is not supported and when the UnidirectionalTopicOperator feature gate is not enabled, the spec.entityOperator.topicOperator property must be removed from the Kafka custom resource.
> - JBOD storage is not supported. The type: jbod storage can be used, but the JBOD array can contain only one disk.

On the other hand, the latest release notes say this (emphasis mine)

> This release contains the following new features and improvements:
> - Add support for Apache Kafka 3.5.2 and 3.6.1
> - The StableConnectIdentities feature gate moves to GA stage and is now permanently enabled without the possibility to disable it.
> - All Connect and Mirror Maker 2 operands will now use StrimziPodSets.
> - The KafkaNodePools feature gate moves to the beta stage and is enabled by default.
> - If needed, KafkaNodePools can be disabled in the feature gates configuration in the Cluster Operator.
> - The UnidirectionalTopicOperator feature gate moves to the beta stage and is enabled by default.
> - If needed, UnidirectionalTopicOperator can be disabled in the feature gates configuration in the Cluster Operator.
> - Improved Kafka Connect metrics and dashboard example files
> - Allow specifying and managing KRaft metadata version
> - **Add support for KRaft to KRaft upgrades (Apache Kafka upgrades for the KRaft-based clusters)**
> - Improved Kafka Mirror Maker 2 dashboard example file
> - **Support for rolling updates of KRaft controller nodes**

Just wanted to check on which of these is more up to date, and in the case of upgrades check if the release notes are correct if upgrading the operator is working or just kafka upgrades themselves.

### Documentation Link

https://strimzi.io/docs/operators/latest/deploying#ref-operator-use-kraft-feature-gate-str

**Comment by scholzj** [comment, marked as answer]
The list in the docs was not updated, the changes from the release notes are correct. Sorry for that.

**Reply by Pythoner6** [reply]
Thanks! That's kind of what I figured, just wanted to double check. What about operator upgrades? The docs mentioned that as an issue and the release notes I think seem to only be about Kafka upgrades?

**Reply by scholzj** [reply]
KRaft upgrades are supported since 0.39.0 on. That includes upgrades to a new Kafka version as well as upgrades of the operator itself.

**Reply by Pythoner6** [reply]
Awesome, thanks!
